src/app/shared/users.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import { UserList } from './users.mock';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UsersService {
+
+  constructor() { }
+
+  getAllUsers() {
+    return UserList;
+  }
+
+  getActiveUsers() {
+    return UserList.filter(function (e) {
+      return e.isDeleted === false;
+    });
+  }
+
+  getDeletedUsers() {
+    return UserList.filter(function (e) {
+      return e.isDeleted === true;
+    });
+  }
+
+  find(id) {
+    return UserList[id - 1];
```

Comment:
Mention the type for the `id` parameter. And may I know the logic behind this statement `usersList[id - 1]`. Why are you subtracting `1` from the `id` ?

Also if you get the `userId: number` as input, then you can simply use the `find` helper method on the `usersList` array

```
    getUserDetails(userId: number) {
        return usersList.find(user => user.id === userId);
    }
```

Reply:
Removed the find method, and used the getUserDetails method instead of that